adal-analytics/src/main/java/com/massivedisaster/adal/analytics/AnalyticsManager.java

```diff
@@ -178,4 +182,97 @@ public void sendEvent(int categoryId, int actionId, String label) {
 
         mTracker.send(builder.build());
     }
+
+    /**
+     * Send an product impression event
+     *
+     * @param product    the product
+     * @param label      the label
+     * @param screenName the screenName
+     */
+    public void sendImpression(Product product, String label, String screenName) {
+        if (getTracker() == null || product == null) {
+            return;
+        }
+
+        HitBuilders.ScreenViewBuilder builder = new HitBuilders.ScreenViewBuilder()
+                .addImpression(product, label);
+
+        mTracker.setScreenName(screenName);
+        mTracker.send(builder.build());
+    }
+
+    /**
+     * Send an product action event
+     *
+     * @param action     the action
+     * @param product    the product
+     * @param screenName the screenName
+     */
+    public void sendProduct(@NonNull ProductAction action, @Nullable Product product, String screenName) {
+        if (getTracker() == null) {
+            return;
+        }
+
+        HitBuilders.ScreenViewBuilder builder = new HitBuilders.ScreenViewBuilder();
+        if (product != null) {
```

Comment:
this if checks nullity for product but in the other methods you didn't check for parameter nullity. Is this really necessary here?